simulator/infra/ports/module.h

```diff
@@ -42,6 +50,8 @@ class Module : public Log
     Module* const parent;
     std::vector<Module*> children;
     const std::string name;
+    std::set<std::string> topology_write_ports;
+    std::set<std::string> topology_read_ports;
```

Comment:
What's wrong with std::vector?

Reply:
Just to be clear: questions do not mean the solution is bad. I just want to know the motivation of the change.

Reply:
It was created to contain unique keys, but there was already protection against duplicates. So <set> is unnecessary.
https://github.com/MIPT-ILab/mipt-mips/blob/59716779f5e86daf64400dbba567e7a1a11b4f2e/simulator/infra/ports/module.cpp#L61-L62
https://github.com/MIPT-ILab/mipt-mips/blob/59716779f5e86daf64400dbba567e7a1a11b4f2e/simulator/infra/ports/module.cpp#L53-L54